code-challenges.test.js

```diff
@@ -14,38 +14,143 @@
+// Convert string back into array use .split method. 
+// Return desired out. 
+
+// const sentenceMaker= (arraySentence) => {
+//   arraySentence.map
+//   arraySentence.join
+//   arraySentence.charAt(0).toUpperCase()
+//   arraySentence.split
+// }
+
+
+//* RAN TEST (RESULTS BELOW): *//
+
+// TypeError: arraySentence.charAt is not a function
+
+// ** NOTES TO INSTRUCTOR **: I know there's not alot of code but this one had me very confused spent multiple hours on it and stackoverflow /Google made me even more lost on how to approach the solution. 
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+// describe("remainderThree", () => {
+//   it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+//     const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+//     // Expected output: [ 2, 0, -1, 0 ]
+//     const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+//     // Expected output: [ 2, 1, -1 ]
+//       expect(remainderThree(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+//       expect(remainderThree(hodgepodge2)).toEqual([ 2, 1, -1 ])
+
+//   })
+// })
 
 
+//* RAN TEST (RESULTS BELOW): *//
+
+// ReferenceError: remainderThree is not defined
+
 // b) Create the function that makes the test pass.
 
+//Goal: Takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3
+
+//Pseudocode:
+// Create a function expression remainderThree. 
+// Create a variable that invokes a for loop parameters on array (array). 
+  // Use .push method to move integers divisiable by 3 to front of array. 
+  // Use .pop to remove Boolean values and string values 
+//Run the function and return expected output. 
+
+// const remainderThree= (array) => {
+//   for(let i=0; i<array.length; i++) {
+//     if(array[i] === "integer") {
+//       array.push(array[i] % 3)
+//       return array 
+// # originally on line #97: ... === typeOf Boolean
+//     } else if(array[i] === 'true' & 'false')
+//       array.pop(array[i])
+//     return array 
+//   } else {
+//     return array.pop(array[i])
+//   }
+// }
+
+//* RAN TEST (RESULTS BELOW): *//
+// SyntaxError: /Users/learnacademy/Desktop/week-6-assessment-scharlez/code-challenges.test.js: Unexpected token (101:2)
+
+// ** NOTES TO INSTRUCTOR **: That syntax error is frustrating, I have rearranged the lines and nothing has fixed it, not to mention the code I'm not sure about but I wanted to see if I could use for loops to solve it. 
```

Comment:
Hi Steven,  your error is because of  a missing { at the end of your else if statement (line 98 above).  I appreciate your hard work to use a for loop.  In this case HOFs will be more useful.  Particularly, .filter() to get the only the integers.